### PR TITLE
chance: init at 4.0.0

### PR DESCRIPTION
**Pull request by Guanran928**
## Description of changes

https://flathub.org/apps/dev.zelikos.rollit
https://gitlab.com/zelikos/rollit

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ch/chance/package.nix [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    owner = "zelikos";
+    repo = "rollit";
+    rev = finalAttrs.version;
+    hash = "sha256-2lZ7iMHMFE1wTSlJj0mIUV62jO0NundYiOC8rdUJGkQ=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit (finalAttrs) src;
+    name = "${finalAttrs.pname}-${finalAttrs.version}";
+    hash = "sha256-7klyWhV9ih9OzJ9JYWLWGHSR9shBHQMBRN+03QuY7bg=";
```
Comment (getchoo) [inline pkgs/by-name/ch/chance/package.nix:35]:
```suggestion
    hash = "sha256-wVZ97dRUozyxOO3ey0Ieom6/y+AEYlBkgWccmRHEm7k=";
```

File: pkgs/by-name/ch/chance/package.nix [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  buildInputs = [
+    cairo
+    gdk-pixbuf
+    glib
+    gtk4
+    libadwaita
+    pango
+  ];
+
```
Comment (getchoo) [inline pkgs/by-name/ch/chance/package.nix:58]:
```suggestion

  passthru.updateScript = nix-update-script { };

```
should help in maintenance

File: pkgs/by-name/ch/chance/package.nix [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  cairo,
+  cargo,
+  desktop-file-utils,
+  fetchFromGitLab,
+  gdk-pixbuf,
+  glib,
+  gtk4,
+  libadwaita,
+  meson,
+  ninja,
```
Comment (getchoo) [inline pkgs/by-name/ch/chance/package.nix:13]:
```suggestion
  ninja,
  nix-update-script,
```

**Review by getchoo — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ch/chance/package.nix (hunk 1 above)
Comment (Guanran928) [reply pkgs/by-name/ch/chance/package.nix:35]: Huh, that's weird

File: pkgs/by-name/ch/chance/package.nix [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  src = fetchFromGitLab {
+    owner = "zelikos";
+    repo = "rollit";
+    rev = finalAttrs.version;
+    hash = "sha256-2lZ7iMHMFE1wTSlJj0mIUV62jO0NundYiOC8rdUJGkQ=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit (finalAttrs) src;
+    name = "${finalAttrs.pname}-${finalAttrs.version}";
```
Comment (Aleksanaa) [inline pkgs/by-name/ch/chance/package.nix:35]:
```suggestion
    name = "chance-${finalAttrs.version}";
```

File: pkgs/by-name/ch/chance/package.nix [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    glib
+    gtk4
+    libadwaita
+    pango
+  ];
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    description = "Roll the dice";
```
Comment (Aleksanaa) [inline pkgs/by-name/ch/chance/package.nix:63]:
```suggestion
    description = "Dice roller built using GTK4 and libadwaita";
```

File: pkgs/by-name/ch/chance/package.nix [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  src = fetchFromGitLab {
+    owner = "zelikos";
+    repo = "rollit";
+    rev = finalAttrs.version;
+    hash = "sha256-2lZ7iMHMFE1wTSlJj0mIUV62jO0NundYiOC8rdUJGkQ=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit (finalAttrs) src;
+    name = "chance-${finalAttrs.version}";
```
Comment (getchoo) [inline pkgs/by-name/ch/chance/package.nix:30]:
```suggestion
    inherit (finalAttrs) pname version src;
```

**Review by getchoo — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ch/chance/package.nix (hunk 6 above)
Comment (Guanran928) [reply pkgs/by-name/ch/chance/package.nix:30]: It's a pretty common pattern, @Aleksanaa treewide pull request soon? :P 
https://github.com/search?q=repo%3ANixOS%2Fnixpkgs+cargoDeps+%3D+rustPlatform.fetchCargoTarball+name+%3D+%22%24%7Bpname%7D-%24%7Bversion%7D%22&type=code

File: pkgs/by-name/ch/chance/package.nix [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+  src = fetchFromGitLab {
+    owner = "zelikos";
+    repo = "rollit";
+    rev = finalAttrs.version;
+    hash = "sha256-2lZ7iMHMFE1wTSlJj0mIUV62jO0NundYiOC8rdUJGkQ=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit (finalAttrs) pname version src;
+    hash = "sha256-wVZ97dRUozyxOO3ey0Ieom6/y+AEYlBkgWccmRHEm7k=";
```
Comment (getchoo) [inline pkgs/by-name/ch/chance/package.nix:30]:
```suggestion
    hash = "sha256-Q4CfDQxlhspjg7Et+0zHwZ/iSnp0CnwwpW/gT7htlL8=";
```

**Review by getchoo — CHANGES_REQUESTED** [review]

**Comment by JohnRTitor** [comment]
Result of `nixpkgs-review pr 324090` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>chance</li>
  </ul>
</details>

**Review by JohnRTitor — APPROVED** [review]

**Comment by JohnRTitor** [comment]
Result of `nixpkgs-review pr 324090` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>chance</li>
  </ul>
</details>

File: pkgs/by-name/ch/chance/package.nix [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+  buildInputs = [
+    libadwaita
+  ];
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    description = "Dice roller built using GTK4 and libadwaita";
+    homepage = "https://gitlab.com/zelikos/rollit";
```
Comment (getchoo) [inline pkgs/by-name/ch/chance/package.nix:53]:
```suggestion
    homepage = "https://gitlab.com/zelikos/rollit";
    changelog = "https://gitlab.com/zelikos/rollit/-/releases/${finalAttrs.version}";
```

**Review by getchoo — CHANGES_REQUESTED** [review]

**Review by getchoo — APPROVED** [review]

**Pull request merged**